### PR TITLE
Don't use custom serialization when reporting type validation problems

### LINKED ISSUE #29188: Get rid of custom serialization when reporting task validation problems via the Problems API
We use `ValidationProblemSerialization` to transfer problems from the type validation worker to the daemon via a json file. Since now we have worker support in the problems API, it should be possible to get rid of the custom serialization code.

### PR DESCRIPTION
**Pull request by donat**
Fixes #29188

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/core-configuration/model-core/src/main/java/org/gradle/internal/isolated/IsolationScheme.java [hunk 1]
```diff
@@ -238,8 +238,8 @@ public Object find(Type serviceType) throws ServiceLookupException {
                 if (serviceClass.isAssignableFrom(BuildServiceRegistry.class)) {
                     return allServices.find(BuildServiceRegistry.class);
                 }
-                if (serviceClass.isAssignableFrom(Problems.class)) {
-                    return allServices.find(Problems.class);
+                if (serviceClass.isAssignableFrom(InternalProblems.class)) {
+                    return allServices.find(InternalProblems.class);
```
Comment (donat) [inline platforms/core-configuration/model-core/src/main/java/org/gradle/internal/isolated/IsolationScheme.java:242]: Thank you for looking into this @hegyibalint!

Comment (hegyibalint) [reply platforms/core-configuration/model-core/src/main/java/org/gradle/internal/isolated/IsolationScheme.java:242]: Trivial solution, isn't it?!... 😂

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this\

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/extensibility/plugin-development/build.gradle.kts [hunk 2]
```diff
@@ -24,13 +24,13 @@ dependencies {
     api(project(":resources"))
     api(project(":toolchains-jvm-shared"))
     api(project(":workers"))
+    api(project(":build-operations"))
+    api(projects.serviceProvider)
```
Comment (donat) [inline platforms/extensibility/plugin-development/build.gradle.kts:28]: SanityCheck failed wo/this change.

File: platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/AbstractPluginValidationIntegrationSpec.groovy [hunk 3]
```diff
@@ -133,61 +137,58 @@ abstract class AbstractPluginValidationIntegrationSpec extends AbstractIntegrati
...
-                    'Add an input or output annotation',
-                    'Mark it as @Internal',
-                ]
-                additionalData.asMap == [
-                    'typeName' : 'MyTask',
-                    'propertyName' : 'ter',
-                ]
-            }
-
+        verifyAll(receivedProblem(0)) {
```
Comment (donat) [inline platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/AbstractPluginValidationIntegrationSpec.groovy:140]: I only removed the `if` statement.

File: platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/CommonPluginValidationIntegrationTest.groovy [hunk 4]
```diff
@@ -16,6 +16,6 @@
 
 package org.gradle.plugin.devel.tasks
 
-class CommonPluginValidationIntegrationTest extends AbstractPluginValidationIntegrationSpec implements ValitdatePluginsTrait {
+class CommonPluginValidationIntegrationTest extends AbstractPluginValidationIntegrationSpec implements ValidatePluginsTrait {
```
Comment (donat) [inline platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/CommonPluginValidationIntegrationTest.groovy:19]: Typo fixed.

File: platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsTrait.groovy [hunk 5]
```diff
@@ -55,16 +54,7 @@ trait ValitdatePluginsTrait implements CommonPluginValidationTrait {
...
-        fails "validatePlugins"
-        def report = new TaskValidationReportFixture(file("build/reports/plugin-development/validation-report.json"))
-        report.verify(messages.collectEntries {
-            def fullMessage = it.message
-            if (!it.defaultDocLink) {
-                fullMessage = "${fullMessage}\n${learnAt(it.id, it.section)}"
-            }
-            [(fullMessage): it.severity]
-        })
-
```
Comment (donat) [inline platforms/extensibility/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsTrait.groovy:67]: We only have `collectedProblems` here.

File: platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java [hunk 6]
```diff
@@ -109,9 +109,10 @@ public void validateTaskClasses() throws IOException {
             });
         getWorkerExecutor().await();
 
-        List<? extends Problem> problems = ValidationProblemSerialization.parseMessageList(new String(Files.readAllBytes(getOutputFile().get().getAsFile().toPath())));
+        Long operationId = Long.valueOf(new String(Files.readAllBytes(getOutputFile().get().getAsFile().toPath())));
```
Comment (donat) [inline platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java:112]: We still need the file here. Now, we only store the current operation ID in there. The daemon can use it to look up which problems were reported from a validation worker.

**Comment by donat** [comment]
Still relevant

**Pull request closed without merging**